### PR TITLE
pip -> uv

### PR DESCRIPTION
**Pull request by gabrieltseng**
- [x] requirements*.txt -> pyproject.toml
- [x] github actions working (test suite takes 3 minutes to run instead of 5!)
- [x] dockerfile updated
- [x] internal updated
- [x] successful test run: https://beaker.allen.ai/orgs/ai2/workspaces/earth-systems/work/01K86J5BSMK3D6B9BYRNHJJA47?taskId=01K86J5BT56QH2GJHY4TNKN6W7&jobId=01K86J5BY58PKSB5B9AGJ1S2SR
- [x] Update readme

This incorporates the changes in #393 .

Switching to uv makes deploying to pypi very easy so I'm also using this PR to document some thoughts I have around the deployment.

---

> [!NOTE]
> Replace requirements/pip with uv and pyproject.toml, updating CI workflows, Dockerfile, and docs; minor internal code adjustments.
> 
> - **Build/Packaging**:
>   - Migrate from `pip`/`requirements*.txt` to `uv` with `pyproject.toml` and `uv.lock`.
> - **CI**:
>   - Update GitHub Actions workflows (`.github/workflows/lint.yaml`, `tests.yaml`) to use `uv` for install and caching.
> - **Docker**:
>   - Update `Dockerfile` to build/install with `uv`.
> - **Tooling**:
>   - Update `.pre-commit-config.yaml`.
> - **Code**:
>   - Adjust internals in `olmoearth_pretrain/internal/common.py` to align with new packaging/setup.
> - **Docs**:
>   - Refresh `README.md` with new install/build/deploy instructions.
> - **Deps files**:
>   - Update/remove legacy `requirements*.txt` files and related lockfiles.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 041c85ec7b13976f0e44b7223a9a608d5fd92b9c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -12,8 +12,26 @@ authors = [
...
+    "pandas>=2.2",
+    "psutil>=7.1.1",
+    "rasterio>=1.4.3",
+    "shapely>=2",
+    "torch>=2.7,<2.8",
+    "torchvision>=0.22.1",
+    "universal-pathlib>=0.2.5",
+    "wandb>=0.19.0",
+    "ai2-olmo-core==2.3.0"
+]
```
[the remark this answers is not in the document]
Comment (gabrieltseng) [reply pyproject.toml:29]: fixed - thanks cursor

File: Dockerfile [hunk 2]
```diff
@@ -6,19 +6,18 @@ FROM ghcr.io/allenai/pytorch:2.5.1-cuda12.1-python3.11
...
-RUN pip install --no-cache-dir -r requirements.txt
+COPY pyproject.toml /stage/pyproject.toml
+COPY uv.lock /stage/uv.lock
+RUN uv sync --all-groups --no-install-project
 
 # Copy the folder `scripts` to `scripts/`
 # You might need multiple of these statements to copy all the folders you need for your experiment.
-COPY helios/ /stage/helios/
-COPY scripts/latent_mim.py /stage/latent_mim.py
-COPY scripts/galileo.py /stage/galileo.py
```
Comment (gabrieltseng) [inline Dockerfile:23]: Do we use this anymore? these lines make me think no

File: pyproject.toml
Comment (gabrieltseng) [inline pyproject.toml:3]: Do we want to bump the version to 1.0.0 for the release?

Comment (gabrieltseng) [inline pyproject.toml]: rslearn and olmoearth_run have as author(s) "the OlmoEarth team" - should we do the same here?

```
authors = [
    { name = "OlmoEarth Team" },
]
```

File: pyproject.toml [hunk 3]
```diff
@@ -12,8 +12,25 @@ authors = [
 readme = "README.md"
 license = {file = "LICENSE"}
```
Comment (gabrieltseng) [inline pyproject.toml:9]: We don't actually have a license file here - I am assuming we follow rslearn and olmoearth_run in adopting Apache 2.0 ? (@pbeukema )

File: pyproject.toml [hunk 4]
```diff
@@ -53,6 +60,65 @@ addopts = [
...
+    "gcsfs>=2025.9.0",
+    "geobench>=1.0.1",
+    "interrogate>=1.7.0",
+    "numpy>=1.26.4",
+    "pandas>=2.3.3",
+    "pandera>=0.26.1",
+    "pytest>=8.4.2",
+    "rasterio>=1.4.3",
+    "rioxarray>=0.19.0",
+    "rslearn==0.0.11",
```
Comment (gabrieltseng) [inline pyproject.toml:86]: just noting we are using the pypi version of rslearn (pinned to the most recent release) instead of the git version now.

File: olmoearth_pretrain/train/loss.py [hunk 5]
```diff
@@ -754,7 +754,7 @@ def compute(
         targets = F.normalize(targets, p=2, dim=-1)
         logits = predictions @ targets.transpose(-2, -1)
 
-        logger.warning(logits.shape)
+        logger.debug(logits.shape)
```
Comment (Hgherzog) [inline olmoearth_pretrain/train/loss.py]: nit: just delete this

File: Dockerfile (hunk 2 above)
Comment (Hgherzog) [reply Dockerfile:23]: this dockerfile is not used AFAIK. We use an olmo-core one for training. I would just delete this because it could be misleading

File: pyproject.toml
Comment (Hgherzog) [reply pyproject.toml]: I think that is appropriate and easier to maintain

File: README.md [hunk 6]
```diff
@@ -7,11 +7,11 @@ Earth system foundation model: data, training, and evaluation
 launching training runs on beaker
 ## General Setup
 
-1. Create a virtual environment in prefered directory with python 3.12 `python3.12 -m venv .venv-olmoearth_pretrain`
-2. Activate the virtual environment `source .venv-olmoearth_pretrain/bin/activate`
-3. Navigate to root directory of this repo and run `pip install -e .`
-4. Run `pip install pre-commit`
-5. Run `pre-commit install`
+1. Install uv (there are some ways to do it [here](https://docs.astral.sh/uv/getting-started/installation/))
```
Comment (Hgherzog) [inline README.md]: nit: add a common command that is easy to copy here

File: README.md [hunk 7]
```diff
@@ -7,11 +7,11 @@ Earth system foundation model: data, training, and evaluation
...
+2. Navigate to root directory of this repo and run `uv sync --locked --all-groups`
+3. Install the pre-commit tool `uv tool install pre-commit --with pre-commit-uv --force-reinstall`
+
+uv installs everything into a venv, so to keep using `python` commands you can activate uv's venv. Otherwise, swap to `uv run python`.
```
Comment (Hgherzog) [inline README.md]: nit: Add `source .venv/bin/activate` as the command for activating the venv. Otherwise it might not be clear that uv automatically creats a venv when you run uv sync.

File: Dockerfile (hunk 2 above)
Comment (gabrieltseng) [reply Dockerfile:23]: removed

File: README.md (hunk 7 above)
Comment (gabrieltseng) [reply README.md]: added

File: README.md (hunk 6 above)
Comment (gabrieltseng) [reply README.md]: added

File: pyproject.toml
Comment (gabrieltseng) [reply pyproject.toml]: updated
